### PR TITLE
DDL: Provide a tunable to control automatic dropping of dependent objects on DROP

### PR DESCRIPTION
**Pull request by nirbhayc**
https://github.com/bloomberg/comdb2/issues/857

### CODE REVIEW

**Comment by adizaimi** [comment]
I am a bit confused from the description in https://github.com/bloomberg/comdb2/issues/857 and the title. From the code change it seems we disallow to drop the column if it is part of an index, is that what this change is really about?

**Comment by nirbhayc** [comment]
> I am a bit confused from the description in #857 and the title. From the code change it seems we disallow to drop the column if it is part of an index, is that what this change is really about?

Yes. There was a typo in the description. Thanks for pointing out.

**Comment by dorinhogea** [comment]
It seems that user needs to run a bunch of DDL in one transaction now to remove a column, I am not sure about this.

**Comment by mponomar** [comment]
Doesn't have to be a transaction.  I think it's better to make the changes explicit.

**Comment by nirbhayc** [comment]
> It seems that user needs to run a bunch of DDL in one transaction now to remove a column, I am not sure about this.

In ALTER TABLE multiple actions can be chained in a single command.

**Comment by adizaimi** [comment]
What would that ALTER look like?

**Comment by nirbhayc** [comment]
> What would that ALTER look like?

Checkout the test in this PR

**Comment by nirbhayc** [comment]
Thanks @mponomar !

File: db/config.c [hunk 1]
```diff
@@ -358,6 +358,7 @@ static void add_legacy_default_options(struct dbenv *dbenv)
         "on accept_on_child_nets",
         "env_messages",
         "off return_long_column_names",
+        "ddl_cascade_drop 0"
```
Comment (ldennis) [inline db/config.c]: Missing `,`

**Comment by nirbhayc** [comment]
Thanks @ldennis !

**Pull request merged**